$NetBSD: patch-ab,v 1.3 2010/06/27 08:49:32 obache Exp $

* GtkSpinButton: setting an adjustment with non-zero page size is deprecated
* support gucharmap-2

--- src/scim_tomoe_prefs_widget.cpp.orig	2007-05-07 07:49:15.000000000 +0000
+++ src/scim_tomoe_prefs_widget.cpp
@@ -170,7 +170,7 @@ create_spin_button (ScimTomoePrefsWidget
     gtk_widget_show (label);
 
     GtkObject *adj = gtk_adjustment_new (value, min, max, step,
-                                         page_size, page_size);
+                                         page_size, 0.0);
     GtkWidget *spin = gtk_spin_button_new (GTK_ADJUSTMENT (adj), 0.0, 0);
     g_object_set_data (G_OBJECT (spin),
                        OBJECT_DATA_CONFIG_KEY, (gpointer) key);
@@ -472,9 +472,13 @@ apply_config (ScimTomoePrefsWidget *pref
         String (SCIM_TOMOE_CONFIG_UNICODE_TABLE_FONT_DEFAULT));
 
     if (font.length() > 0) {
-        gucharmap_charmap_set_font (charmap, font.c_str ());
+        PangoFontDescription *desc = pango_font_description_from_string(font.c_str ());
+        gucharmap_charmap_set_font_desc (charmap, desc);
+        pango_font_description_free (desc);
     } else {
-        gucharmap_charmap_set_font (charmap, NULL);
+        PangoFontDescription *desc = pango_font_description_new ();
+        gucharmap_charmap_set_font_desc (charmap, desc);
+        pango_font_description_free (desc);
         set_default_font (prefs, SCIM_TOMOE_CONFIG_CAND_FONT,
                           GTK_WIDGET (charmap));
     }
